### PR TITLE
Implemented leadkedCred Detail Api

### PR DESCRIPTION
**Pull request by Sarvesh025**

### CODE REVIEW

File: api/leaked_credentials/cred.controller.js [hunk 1]
```diff
@@ -39,36 +40,36 @@ initializeDb();
...
+  try {
+    //Create a leakCred
+    const leakCred = {
+      id: req.body.id,
+      email: req.body.email,
+      created_at: req.body.created_at,
+      updated_at: req.body.updated_at,
+      created_by: req.body.created_by,
+      updated_by: req.body.updated_by,
+      no_leaks: req.body.no_leaks,
```
Comment (Sarvesh025) [inline api/leaked_credentials/cred.controller.js:59]: Remove this coma.

File: api/leaked_credentials/cred.controller.js [hunk 2]
```diff
@@ -39,36 +40,36 @@ initializeDb();
...
+    };
+    db.models.LeakCred.create(leakCred)
+      .then((data) => {
+        res.status(201).json(data);
+      })
+      .catch((error) => {
         res.status(500).json({ message: error.message });
-        return;
-    }
-};
```
Comment (Sarvesh025) [inline api/leaked_credentials/cred.controller.js:71]: put semiColon

File: api/leaked_credentials/cred.controller.js [hunk 3]
```diff
@@ -90,17 +91,71 @@ async function create (req, res) {
...
+    });
+    const totalCount = await db.models.LeakCred.count();
+    const totalPages = Math.ceil(totalCount / Number(pageSize));
+
+    res.status(200).json({ count, totalPages });
+  } catch (error) {
+    console.error("Error counting leaked emails:", error);
+    res.status(500).json({ error: "Internal server error" });
+  }
+}
```
Comment (Sarvesh025) [inline api/leaked_credentials/cred.controller.js:115]: Put semiColon

File: api/leaked_credentials/cred.controller.js [hunk 4]
```diff
@@ -90,17 +91,71 @@ async function create (req, res) {
...
+        res.status(200).json({ data, totalPages });
+      })
+      .catch((error) => {
         res.status(500).json({ message: error.message });
-    }
+      });
+  } catch (error) {
+    res.status(500).json({ message: error.message });
+  }
+}
```
Comment (Sarvesh025) [inline api/leaked_credentials/cred.controller.js:137]: Put semiColon

File: api/leaked_credentials/cred.controller.js [hunk 5]
```diff
@@ -90,17 +91,71 @@ async function create (req, res) {
...
-        .then((data)=>{
-            res.status(200).json(data);
-        }).catch((error)=>{
-            res.status(500).json({ message: error.message });
-        });
-    } catch (error) {
+
+//This returns total number of leaks per email address.
+const { Op } = Sequelize;
+async function leakCounts(req, res) {
```
Comment (arijeet-mukherjee) [inline api/leaked_credentials/cred.controller.js:97]: This is useless method , it should query the LeakCredDetail table to get all the data where it has been leaked.

Comment (arijeet-mukherjee) [inline api/leaked_credentials/cred.controller.js:97]: rename it to getTotalLeakCounts

File: api/leaked_credentials/credDetail.controller.js [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+        console.log(emails);
+        res.json({ emails, totalPages });
+    } catch (error) {
+        console.error("Error fetching tracked emails:", error);
+        res.status(500).json({ error: "Internal server error" });
+    }
+}
+
+const { Op } = Sequelize;
+async function getByCondition(req, res){
```
Comment (arijeet-mukherjee) [inline api/leaked_credentials/credDetail.controller.js:25]: Try to reuse this method in LeakCred controller to get the total leaks count

File: api/leaked_credentials/credDetail.controller.js [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+      const totalCount = await db.models.LeakCredDetail.count();
+      const totalPages = Math.ceil(totalCount / Number(pageSize));
+  
+      res.status(200).json({ data, totalPages });
+    } catch (error) {
+      console.error("Error counting leaked emails:", error);
+      res.status(500).json({ error: "Internal server error" });
+    }
+}
+
```
Comment (arijeet-mukherjee) [inline api/leaked_credentials/credDetail.controller.js:44]: Implement a method to get the leak cred details as well. I can input either leak cred id or email id

**Pull request closed without merging**